fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/EntityDatatableChecksReadPlatformServiceImpl.java

```diff
@@ -174,20 +174,22 @@ private List<DatatableChecksData> getDataTables() {
         return this.jdbcTemplate.query(sql, this.registerDataTableMapper); // NOSONAR
     }
 
-    protected static final class RegisterDataTableMapper implements RowMapper<DatatableChecksData> {
+    protected final class RegisterDataTableMapper implements RowMapper<DatatableChecksData> {
+
+        public static final String SELECT_FROM = " t.application_table_name as entity, t.registered_table_name as tableName FROM x_registered_table t WHERE ";
 
         @Override
         public DatatableChecksData mapRow(final ResultSet rs, @SuppressWarnings("unused") final int rowNum) throws SQLException {
-
             final String entity = rs.getString("entity");
             final String tableName = rs.getString("tableName");
 
             return new DatatableChecksData(entity, tableName);
         }
 
         public String schema() {
-            return " t.application_table_name as entity, t.registered_table_name as tableName " + " from x_registered_table t "
-                    + " where application_table_name IN( 'm_client','m_group','m_savings_account','m_loan')";
+            String[] values = EntityTables.getFiltered(EntityTables::hasCheck).stream().map(EntityTables::getName).toArray(String[]::new);
+            return SELECT_FROM + SqlOperator.IN.formatSql(databaseTypeResolver.databaseType(), JdbcJavaType.VARCHAR,
```

Comment:
This SqlOperator doesn't seem like to be the best idea.. very confusing to use. Why don't we simply do the "IN" condition as a string here? Nothing seems to hard about that as far as I can see.

Reply:
I do not know what to answer to that use string literals was better. I do not agree at all, but I am open to discuss.